zone/raids.cpp

```diff
@@ -2203,3 +2259,735 @@ void Raid::SetNewRaidLeader(uint32 i)
+		}
+}
+
+void Raid::RemoveRaidDelegates(const char* delegatee)
+{
+	auto ma = members[GetPlayerIndex(delegatee)].main_assister;
+	auto mm = members[GetPlayerIndex(delegatee)].main_marker;
+
+	if (ma) {
+		SendRemoveRaidXTargets(static_cast<XTargetType>(RaidAssist1 + ma - 1));
+		SendRemoveRaidXTargets(static_cast<XTargetType>(RaidAssist1Target + ma - 1));
+		DelegateAbilityAssist(leader->CastToMob(), delegatee);
+	}
+
+	if (mm) {
+		SendRemoveRaidXTargets(static_cast<XTargetType>(RaidMarkTarget1 + mm - 1));
+		DelegateAbilityMark(leader->CastToMob(), delegatee);
+	}
+}
+
+void Raid::SendRemoveAllRaidXTargets(const char* client_name)
+{
+
+	auto c = entity_list.GetClientByName(client_name);
+
+	for (int i = 0; i < c->GetMaxXTargets(); ++i)
+	{
+		if ((c->XTargets[i].Type == RaidAssist1) ||
+			(c->XTargets[i].Type == RaidAssist2) ||
+			(c->XTargets[i].Type == RaidAssist3) ||
+			(c->XTargets[i].Type == RaidAssist1Target) ||
+			(c->XTargets[i].Type == RaidAssist2Target) ||
+			(c->XTargets[i].Type == RaidAssist3Target) ||
+			(c->XTargets[i].Type == RaidMarkTarget1) ||
+			(c->XTargets[i].Type == RaidMarkTarget2) ||
+			(c->XTargets[i].Type == RaidMarkTarget3))
+		{
+			c->XTargets[i].ID = 0;
+			c->XTargets[i].Name[0] = 0;
+			c->SendXTargetPacket(i, nullptr);
+		}
+	}
+}
+
+void Raid::SendRemoveRaidXTargets(XTargetType Type)
+{
+
+	for (const auto& m : members) {
+		if (m.member && !m.is_bot) {
+			for (int i = 0; i < m.member->GetMaxXTargets(); ++i)
+			{
+				if (m.member->XTargets[i].Type == Type)
+				{
+					m.member->XTargets[i].ID = 0;
+					m.member->XTargets[i].Name[0] = 0;
+					m.member->SendXTargetPacket(i, nullptr);
+				}
+			}
+		}
+	}
+}
+
+void Raid::SendRemoveAllRaidXTargets()
+{
+	for (const auto& m : members) {
+		if (m.member && !m.is_bot) {
+			for (int i = 0; i < m.member->GetMaxXTargets(); ++i)
+			{
+				if ((m.member->XTargets[i].Type == RaidAssist1) ||
+					(m.member->XTargets[i].Type == RaidAssist2) ||
+					(m.member->XTargets[i].Type == RaidAssist3) ||
+					(m.member->XTargets[i].Type == RaidAssist1Target) ||
+					(m.member->XTargets[i].Type == RaidAssist2Target) ||
+					(m.member->XTargets[i].Type == RaidAssist3Target) ||
+					(m.member->XTargets[i].Type == RaidMarkTarget1) ||
+					(m.member->XTargets[i].Type == RaidMarkTarget2) ||
+					(m.member->XTargets[i].Type == RaidMarkTarget3))
+				{
+					m.member->XTargets[i].ID = 0;
+					m.member->XTargets[i].Name[0] = 0;
+					m.member->SendXTargetPacket(i, nullptr);
+				}
+			}
+		}
+	}
+}
+
+void Raid::SendRaidAssistTarget()
+{
+	// Send a packet to the entire raid notifying them of the group target selected by the Main Assist.
+
+	uint16 assist_target_id = 0;
+	uint16 number = 0;
+	Mob* target = nullptr;
+
+	if (strlen(main_assister_pcs[MAIN_ASSIST_1_SLOT]) > 0)
+	{
+		auto player = entity_list.GetMob(main_assister_pcs[MAIN_ASSIST_1_SLOT]);
+		if (player) {
+			target = entity_list.GetMob(main_assister_pcs[MAIN_ASSIST_1_SLOT])->GetTarget();
```

Comment:
This can actually cause a crash, we need to be validating `GetMob` is a valid pointer before further accessing it with `GetTarget`

Reply:
Same for other places where you're also doing the same

Reply:
Looking at your code this should just be `target = player->GetTarget()`

Reply:
Isn't it done via the player check?

Reply:
Great point.  It certainly can.  I will let you review this evening as past my bedtime.  I will update tomorrow and any other suggestions.  Thanks again for assist.

Reply:
I have updated this.  I believe this is the last outstanding review item.  Let me know if there is anything else that I can help with.